### ISSUE TITLE
upload speed reported different between version 0.3.4 and 0.3.1

### ISSUE THREAD

**Issue by sebastien-bratieres** [sivel/speedtest-cli#198]
(you might want to wontfix this in case you judge it has to do with inconsistency; however, in this particular case, different versions of speedtest-cli return inconsistent values so it might be worth looking into it; it might also be related to #148 )

(on Windows 8, Python 3.4.3, Miniconda3 install)

```
hockey@GAWLER ~
$ pip install speedtest-cli==0.3.1
Collecting speedtest-cli==0.3.1
  Downloading speedtest-cli-0.3.1.tar.gz
Building wheels for collected packages: speedtest-cli
  Running setup.py bdist_wheel for speedtest-cli
  Stored in directory: C:\Users\hockey\AppData\Local\pip\Cache\wheels\c7\52\ca\447dcf1dcddff329c4f1eaef7a0866cc1b0983a30f9d271302
Successfully built speedtest-cli
Installing collected packages: speedtest-cli
  Found existing installation: speedtest-cli 0.3.4
    Uninstalling speedtest-cli-0.3.4:
      Successfully uninstalled speedtest-cli-0.3.4
Successfully installed speedtest-cli-0.3.1

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 33.592 ms
Download: 5.37 Mbits/s
Upload: 0.98 Mbits/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 27.314 ms
Download: 5.57 Mbits/s
Upload: 0.96 Mbits/s

hockey@GAWLER ~
$ pip install speedtest-cli==0.3.4
Collecting speedtest-cli==0.3.4
  Using cached speedtest_cli-0.3.4-py2.py3-none-any.whl
Installing collected packages: speedtest-cli
  Found existing installation: speedtest-cli 0.3.1
    Uninstalling speedtest-cli-0.3.1:
      Successfully uninstalled speedtest-cli-0.3.1
Successfully installed speedtest-cli-0.3.4

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 30.181 ms
Download: 5.43 Mbit/s
Upload: 0.00 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 28.152 ms
Download: 5.54 Mbit/s
Upload: 0.20 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 34.025 ms
Download: 5.55 Mbit/s
Upload: 0.20 Mbit/s

hockey@GAWLER ~
$ pip install speedtest-cli==0.3.1
Collecting speedtest-cli==0.3.1
Installing collected packages: speedtest-cli
  Found existing installation: speedtest-cli 0.3.4
    Uninstalling speedtest-cli-0.3.4:
      Successfully uninstalled speedtest-cli-0.3.4
Successfully installed speedtest-cli-0.3.1

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 27.957 ms
Download: 5.52 Mbits/s
Upload: 0.98 Mbits/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 35.864 ms
Download: 5.56 Mbits/s
Upload: 0.98 Mbits/s
```

**Comment by sivel** [comment]
The only real difference that I could remotely attribute this to, is that we are always sending a user agent now, whereas before we were not.  Additionally there is some code added around catching exceptions in http requests that was added.

Seeing as though the download seems to be consistent makes this kinda strange.  If you feel like testing more, you could also test with 0.3.2 and 0.3.3 and see where the issue starts at.

I cannot duplicate or estimate what the issue is.

If you perform the additional testing, just comment, and I'll look into it.  Otherwise I'll close this issue in the future.

**Comment by sebastien-bratieres** [comment]
I fail to reproduce what I had due to an error with 0.3.1 now:

````
hockey@GAWLER ~
$ pip install speedtest-cli==0.3.1
Collecting speedtest-cli==0.3.1
Installing collected packages: speedtest-cli
  Found existing installation: speedtest-cli 0.3.4
    Uninstalling speedtest-cli-0.3.4:
      Successfully uninstalled speedtest-cli-0.3.4
Successfully installed speedtest-cli-0.3.1

hockey@GAWLER ~
$ speedtest-cli --simple
Failed to parse list of speedtest.net servers
````

**Comment by sebastien-bratieres** [comment]
Tried with 0.3.0 instead of 0.3.1. The difference in upload speed starts to show with 0.3.2.

````
hockey@GAWLER ~
$ pip --quiet install speedtest-cli==0.3.0

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 27.353 ms
Download: 11.28 Mbits/s
Upload: 1.00 Mbits/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 27.506 ms
Download: 11.22 Mbits/s
Upload: 0.96 Mbits/s

hockey@GAWLER ~
$ pip --quiet install speedtest-cli==0.3.1

hockey@GAWLER ~
$ speedtest-cli --simple
Failed to parse list of speedtest.net servers

hockey@GAWLER ~
$ pip --quiet install speedtest-cli==0.3.2

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 28.393 ms
Download: 11.27 Mbit/s
Upload: 0.20 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 28.892 ms
Download: 11.19 Mbit/s
Upload: 0.00 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 28.974 ms
Download: 11.27 Mbit/s
Upload: 0.34 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 28.866 ms
Download: 11.19 Mbit/s
Upload: 0.20 Mbit/s

hockey@GAWLER ~
$ pip --quiet install speedtest-cli==0.3.3

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 30.195 ms
Download: 11.21 Mbit/s
Upload: 0.33 Mbit/s

hockey@GAWLER ~
$ speedtest-cli --simple
Ping: 32.567 ms
Download: 11.26 Mbit/s
Upload: 0.00 Mbit/s
````

**Comment by theAdmiral777** [comment]
I also have been experiencing the same issue. I ran both v0.3.1 and v0.3.2:
```
v0.3.1
Ping: 11.017 ms
Download: 6.85 Mbits/s
Upload: 0.50 Mbits/s

v0.3.2
Ping: 12.137 ms
Download: 3.19 Mbit/s
Upload: 0.00 Mbit/s
```
v0.3.1 consistently gave the correct results and it seems any version after v0.3.1 gave incorrect results.

Windows 8.1 & Python 3.4

**Comment by kerby82** [comment]
I have the same problem on some servers and considering the change in the commit the problem is not a problem. The connection is lasting more than the default timeout. Try to use for example --timeout 300 and you will see that the Upload will be not 0 Mbit/s.

**Comment by sivel** [comment]
Please test with the recently released v1.0.0.
